libraries/AP_GPS/AP_GPS_UAVCAN.cpp

```diff
@@ -141,10 +176,26 @@ AP_GPS_UAVCAN* AP_GPS_UAVCAN::get_uavcan_backend(AP_UAVCAN* ap_uavcan, uint8_t n
             if (_detected_modules[i].ap_uavcan == nullptr) {
                 _detected_modules[i].ap_uavcan = ap_uavcan;
                 _detected_modules[i].node_id = node_id;
+                // Just set the Node ID in order of appearance
+                // This will be used to set select ids
+                AP::gps()._node_id[i].set_and_notify(node_id);
                 break;
             }
         }
     }
+    struct DetectedModules tempslot;
+    // Sort based on the node_id, larger values first
+    // we do this, so that we have repeatable GPS
+    // registration
+    for (uint8_t i = 1; i < GPS_MAX_RECEIVERS; i++) {
```

Comment:
i'd love us to have a standard sort function, preferably with a lambda if possible